packages/http/src/router/matchPath.ts

```diff
@@ -1,18 +1,29 @@
 import { MatchType } from './types';
 import * as E from 'fp-ts/Either';
 
-function fragmentarize(path: string): string[] {
+function fragmentize(path: string): string[] {
   return path.split('/').slice(1);
 }
 
-function getTemplateParamName(pathFragment: string) {
-  const match = /{(.*)}/.exec(pathFragment);
+// Attempt to decode path fragment. Decode should not do any harm since it is
+// decoding only URI sequences which are previously created by encodeURIComponent
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent
+function decodePathFragment(pathFragment?: string) {
+  try {
+    return pathFragment && decodeURIComponent(pathFragment);
+  } catch (_) {
+    return pathFragment;
+  }
+}
+
+function getTemplateParamName(pathFragment?: string) {
+  const match = typeof pathFragment === 'string' && /{(.*)}/.exec(pathFragment);
```

Comment:
I feel like `mathPath` lacks a functional approach. How to turn 2 linkes into 11 lines ;)

```
  return pipe(
    O.fromNullable(pathFragment),
    O.map(pathFragment =>
      pipe(
        pathFragment,
        O.fromPredicate(pathFragment => typeof pathFragment === 'string'),
        O.map((pathFragment: string) => /{(.*)}/.exec(pathFragment)),
        O.map(m => m && m[1])
      )
    )
  );

```

Reply:
Yeah, it definitely is. In this particular case I just fixed type issue, let's plan a functional overhaul of `matchPath.ts` for another time 😉